main.tf

```
@@ -117,13 +116,13 @@ resource aws_route "private_routes" {
 }
 
 resource aws_route_table_association "public_route_association" {
-  count          = "${var.az_count}"
+  count          = "${length(var.public_cidr_ranges)}"
   subnet_id      = "${element(aws_subnet.public_subnet.*.id, count.index)}"
   route_table_id = "${aws_route_table.public_route_table.id}"
 }
 
 resource aws_route_table_association "private_route_association" {
-  count          = "${var.az_count}"
+  count          = "${length(var.private_cidr_ranges)}"
```

This line should probably be since we need one association per subnet, the counts should match.
```suggestion
  count          = "${var.az_count * var.private_subnets_per_az}"
```